deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -28,211 +30,456 @@ class DeepLakeVectorStore:
+            >>> # Create a vector store with custom tensors
+            >>> data = DeepLakeVectorStore(
+            >>>        path = <path_for_storing_data>,
+            >>>        tensor_params = [{"name": "text", "htype": "text"},
+            >>>                         {"name": "embedding_1", "htype": "embedding"},
+            >>>                         {"name": "embedding_2", "htype": "embedding"},
+            >>>                         {"name": "source", "htype": "text"},
+            >>>                         {"name": "metadata", "htype": "json"}
+            >>>                        ]
+            >>> )
 
         Args:
-            dataset_path (str): path to the deeplake dataset. Defaults to DEFAULT_VECTORSTORE_DEEPLAKE_PATH.
-            token (str, optional): Activeloop token, used for fetching credentials for Deep Lake datasets. This is Optional, tokens are normally autogenerated. Defaults to None.
-            embedding_function (Optional[callable], optional): Function that converts query into embedding. Defaults to None.
-            read_only (bool, optional):  Opens dataset in read-only mode if this is passed as True. Defaults to False.
-            ingestion_batch_size (int): The batch size to use during ingestion. Defaults to 1024.
+            path (str, pathlib.Path): - The full path for storing to the Deep Lake Vector Store. It can be:
+                - a Deep Lake cloud path of the form ``hub://org_id/dataset_name``. Requires registration with Deep Lake.
+                - an s3 path of the form ``s3://bucketname/path/to/dataset``. Credentials are required in either the environment or passed to the creds argument.
+                - a local file system path of the form ``./path/to/dataset`` or ``~/path/to/dataset`` or ``path/to/dataset``.
+                - a memory path of the form ``mem://path/to/dataset`` which doesn't save the dataset but keeps it in memory instead. Should be used only for testing as it does not persist.
+            tensor_params (List[Dict[str, dict]], optional): List of dictionaries that contains information about tensors that user wants to create. See `create_tensor` in Deep Lake API docs for more information. Defaults to `DEFAULT_VECTORSTORE_TENSORS`.
+            embedding_function (Optional[callable], optional): Function that converts the embeddable data into embeddings. Defaults to None.
+            read_only (bool, optional):  Opens dataset in read-only mode if True. Defaults to False.
+            ingestion_batch_size (int): Batch size used during ingestion. Defaults to 1024.
             num_workers (int): The number of workers to use for ingesting data in parallel. Defaults to 0.
-            exec_option (str): Type of query execution. It could be either "python", "compute_engine" or "tensor_db". Defaults to "python".
+            exec_option (str): Default method for search execution. It could be either "python", "compute_engine" or "tensor_db". Defaults to "python".
```

Comment:
Why is this called the `Default method..`? Should be just `Method for search execution` imo.